rpc/common.ts

```diff
@@ -1,33 +1,508 @@
+import type { Snowflake } from '../globals';
+import type { APIMessage, APIUser } from '../v10';
+
+/**
+ * @unstable the ping object for the `VOICE_CONNECTION_STATUS` dispatched `pings` field,
+ * but discord's documentation incorrectly documents it as an 'array of integers'.
+ */
+export interface RPCVoiceConnectionStatusPing {
+	/**
+	 * the time the ping was sent
+	 */
+	time: number;
+	/**
+	 * the latency of the ping (in ms)
+	 */
+	value: number;
+}
+
+/**
+ * @unstable
+ */
+export interface RPCAPIMessageParsedContentOriginalMatch {
+	0: string;
+	index: 0;
+}
+
+/**
+ * @unstable
+ */
+export interface RPCAPIMessageParsedContentText {
+	type: 'text';
+	originalMatch: RPCAPIMessageParsedContentOriginalMatch;
+	content: string;
+}
+
+/**
+ * @unstable
+ */
+export interface RPCAPIMessageParsedContentMention {
+	type: 'mention';
+	userId: Snowflake;
+	channelId: Snowflake;
+	guildId: Snowflake;
+	/**
+	 * same as `userId`
+	 */
+	parsedUserId: Snowflake;
```

Comment:
Might be better to type this as `RPCAPIMessageParsedContentMention["userId"]` if it truly is the same.